Add Kyma eventing use cases (#177)
kyma-project/community: sig-and-wg/wg-knative/kyma-eventing-use-case.md

![](assets/extend-application.png)

* Application preregisters the event-schema for the events it wishes to send to Kyma.
* Event schema is defined in [Async API format](https://github.com/asyncapi/asyncapi)
* The events registered by the external solutions can be updated. This implies event schemas can change, new event schemas can be added or existing ones can be removed. Such changes need to be cascaded to the underlying layer which will be Knative eventing.
* Application developers only deal with event types domain. They have no knowledge or coupling with the underlying messaging system.
* Events are stored centrally. Event Bus is a core component.
* Event origin is defined using identifiers and the identifiers are used for the data partition. e.g. `customer.created` event from `Application X` is segregated from `customer.created` event from `Application Y`.

**Integrating Applications**

Event-driven computations in Kyma can also be used to integrate two or more applications in the customer ecosystem.

Events produced by `Application X` can be used to trigger workflows in `Application Y` via APIs and vice-versa.

# Use case 2: Asynchronous workflows

> Application developers can build serverless applications in Kyma which produce internal events. These events can be used to trigger serverless applications that consume these events thereby creating an asynchronous pipeline or workflow.

Reviewer: ```suggestion
Application developers can build in Kyma serverless applications which produce internal Events. These Events can be used to trigger serverless applications that consume the Events. This way, you create an asynchronous pipeline or workflow.
```

![](assets/asynchronous-workflows.png)

# Use case 3: Integrating third-party messaging systems

> Kyma can be used to integrate third-party messaging systems (including cloud solutions such as `Google PubSub`). A business happening in the application can be captured as an event in Kyma. This event can then be stored in the cloud eventing system such as `Google PubSub`. Based on the event, serverless computations can be triggered in GCP to achieve a business scenario.

Reviewer: ```suggestion
Kyma can be used to integrate third-party messaging systems, including cloud solutions such as `Google PubSub`. A business happening in the application can be captured as an Event in Kyma. This event can then be stored in a cloud eventing system such as `Google PubSub`. Based on the Event, serverless computations can be triggered in GCP to achieve a given business scenario.
```

![](assets/integrate-3rd-party-publish.png)

> An event in the third party messaging system such as `Google PubSub` can be captured by Kyma to trigger a business action in the application.

Reviewer: ```suggestion
An Event in a third-party messaging system can be captured by Kyma to trigger a certain business logic in the application.
```
